portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -22,25 +28,42 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+/** Servlet that handles comments data using Datastore */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
   private final ArrayList<String> comments = new ArrayList<String>();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String json = convertToJsonUsingGson(comments);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    PreparedQuery results = datastore.prepare(query);
+
+    for (Entity commentEntity : results.asIterable()) {
+      String comment = (String) commentEntity.getProperty("comment");
+      String name = (String) commentEntity.getProperty("name");
+    }
+
+    String jsonComments = convertToJsonUsingGson(comments);
     response.setContentType("application/json");
-    response.getWriter().println(json);
+    response.getWriter().println(jsonComments);
   }
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     String commentString = request.getParameter("comment-input");
     String nameString = request.getParameter("name-input");
+    long timestamp = System.currentTimeMillis();
```

Comment:
Nit: For timestamps represented by numbers, it's common to include the unit - for example, timestampMs.